stats/chan.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * skogul, internal stats
+ *
+ * Copyright (c) 2021 Telenor Norge AS
+ * Author(s):
+ *  - Håkon Solbjørg <hakon.solbjorg@telenor.no>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+/* stats supports Skogul by adding internal statistics
+   about how Skogul is doing. The stats are very generic
+   and are sent as regular `skogul.Metric`s. By configuring
+   the stats receiver all stats will be picked up and sent
+   through skogul as any other metrics.
+*/
+package stats
+
+import (
+	"context"
+	"time"
+
+	"github.com/telenornms/skogul"
+)
+
+// DefaultInterval is the default interval used for sending stats.
+var DefaultInterval = time.Second * 10
+
+var statsLog = skogul.Logger("stats", "chan")
+
+// StatsChan is a channel which accepts skogul statistic as a skogul.Metric
+// By configuring the stats receiver, this channel is drained and sent on to
+// the specified handler.
+var Chan chan *skogul.Metric
+
+// StatsDrainCtx and StatsDrainCancel are the context and cancel functions
+// for the automatically created stats.StatsChan.
+// If a skogul stats receiver is configured, StatsDrainCancel MUST be called
+// so that statistics are not discarded.
+var StatsDrainCtx, StatsDrainCancel = context.WithCancel(context.Background())
```

Comment:
Same comment with name as I had on the channel earlier. Let's not talk about stats.StatsDrainCtx when we could be talking about stats.DrainCtx :D